cmd/es-rollover/app/init/action.go

```
@@ -69,29 +66,16 @@ func (c Action) Do() error {
 }
 
 func createIndexIfNotExist(c client.IndexAPI, index string) error {
-	err := c.CreateIndex(index)
+	exists, err := c.IndexExists(index)
 	if err != nil {
-		var esErr client.ResponseError
-		if errors.As(err, &esErr) {
-			if esErr.StatusCode != http.StatusBadRequest || esErr.Body == nil {
-				return esErr.Err
-			}
-			// check for the reason of the error
-			jsonError := map[string]any{}
-			err := json.Unmarshal(esErr.Body, &jsonError)
-			if err != nil {
-				// return unmarshal error
-				return err
-			}
-			errorMap := jsonError["error"].(map[string]any)
-			// check for reason, ignore already exist error
-			if strings.Contains(errorMap["type"].(string), "resource_already_exists_exception") {
```

why was wrong with this logic? It seems it was already able to detect cases where an object was already present.

---

There is nothing wrong in logic but this error can be avoided by checking whether index exists or not, along with this it is hard to track other errors which can come due to already existing indices. The current error can also be produced on non-existing indices because alias can point towards a non-existing index. But this is an uncertain error as there is no way to check the alias pointing to a non-existing index. Therefore I thought of eradicating the possible certain errors (like resource existing error) and ignoring these unavoidable errors (alias pointing towards non-existing errors). So we can approach by these two ways:
1. Following the approach in this PR: Checking whether index exists or not and ignoring error for non-existing indices.
2. Following the current approach: Ignoring both errors.

Would be thankful to get your views on choosing one of these approaches!

---

Sounds good, I just wanted to make sure that you actually have good reasons to go in a new direction. Please include the above motivation in the PR description.